File: 0268. Missing Number/0268.py
# time complexity O(n)
# space complexity O(1)
class Solution(object):
    # gauss summation
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # sum = n(n+1)/2
        n = len(nums)
        expected_sum = n*(n+1)//2 # // returns the closest integer
    
        actual_sum = 0
        for num in nums:
            actual_sum += num # total of the numbers in the array

        return expected_sum - actual_sum # return the missing number

File: 0268. Missing Number/0268.java
// time complexity O(n)
// space complexity O(n)
class Solution {
    public int missingNumber(int[] nums) {
        // create array of size length+1 for all elements
        int[] arr = new int[nums.length+1];

        // mark the existing numbers
        for(int i=0; i<nums.length; i++){
            arr[nums[i]]++;
        }

        // find the missing number
        for (int i=0; i<nums.length+1; i++){
            // System.out.print(arr[i] + " ");
            if(arr[i]==0) return i;
        }

        return -1;
    }
}